api/src/main/java/com/exelaration/abstractmemery/services/implementations/ImageServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.exelaration.abstractmemery.services.implementations;
+
+import com.exelaration.abstractmemery.domains.Image;
+import com.exelaration.abstractmemery.services.FileStorageService;
+import com.exelaration.abstractmemery.services.ImageService;
+import com.exelaration.abstractmemery.services.MetadataService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+@Service("imageService")
+public class ImageServiceImpl implements ImageService {
+
+  @Autowired private FileStorageService fileStorageService;
+
+  @Autowired private MetadataService metadataService;
+
+  public Image save(MultipartFile file) {
+    fileStorageService.createDirectory();
```

Comment:
Related to comment about this method not being in the interface, it should be the storage service's job to ensure the file can be saved to the proper place.  The client using it shouldn't have to care if the directory exists or not, just whether or not the file can be saved.